run-tests.sh

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+cd "$(dirname "$0")"
+
+generate_id() {
+  LC_ALL=C tr -dc 'a-z' </dev/urandom | head -c 13 ; echo
+}
+
+port=9000
+containers=()
+sonarqube_container_name="it-sonarqube"
+network="it-network"
+
+print_usage() {
+  cat <<EOF
+usage: $0 [IMAGE...]
+
+examples:
+       $0 "sonar-scanner-cli:4.1"
+EOF
+}
+
+info() {
+  echo "[info] $@"
+}
+
+warn() {
+  echo "[warn] $@" >&2
+}
+
+fatal() {
+  echo "[error] $@" >&2
+  exit 1
+}
+
+require() {
+  local prog missing=()
+  for prog; do
+    if ! type "$prog" &>/dev/null; then
+      missing+=("$prog")
+    fi
+  done
+
+  [[ ${#missing[@]} == 0 ]] || fatal "could not find required programs on the path: ${missing[@]}"
+}
+
+create_network() {
+  docker network create "$network" || true
+}
+
+destroy_network() {
+  docker network rm "$network"
+}
+
+wait_for_sonarqube() {
+  local i web_up=no sonarqube_up=no
+
+  for ((i = 0; i < 10; i++)); do
+    info "waiting for web server to start ..."
+    if curl -sI localhost:$port | grep '^HTTP/.* 200'; then
+      web_up=yes
+      break
+    fi
+    sleep 5
+  done
+
+  [[ $web_up == yes ]] || return 1
+
+  for ((i = 0; i < 20; i++)); do
+    info "waiting for sonarqube to be ready ..."
+    if curl -s localhost:$port/api/system/status | grep '"status":"UP"'; then
+      sonarqube_up=yes
+      break
+    fi
+    sleep 10
+  done
+
+  [[ "$sonarqube_up" == yes ]]
+}
+
+test_scanner() {
+  local scanner_finished_successfuly=no container_name
+  container_name=$(generate_id)
+  info "testing image $1 in container $container_name"
+
+  git clone https://github.com/SonarSource/sonar-scanning-examples.git "$container_name"
+
+  scanner_props_location="$(pwd)/$container_name/sonarqube-scanner/sonar-project.properties"
+  echo "sonar.projectKey=$container_name-test" >> "$scanner_props_location"
+  echo "sonar.host.url=http://${sonarqube_container_name}:9000" >> "$scanner_props_location"
+
+  docker run --network="$network" --name="$container_name" -it -v "$(pwd)/$container_name/sonarqube-scanner:/usr/src" "$1"
+  containers+=("$container_name")
+  docker wait "$container_name"
+  info "Container $container_name stopped."
+  if docker logs "$container_name" | grep 'INFO: EXECUTION SUCCESS'; then
+    scanner_finished_successfuly=yes
+  fi
+
+  rm -rf "$container_name"
```

Comment:
fixed

Reply:
I'm curious and would like to learn: could you share how you fixed it? I couldn't figure it out reading the latest version of the code

Reply:
The container uses non-root user to operate now which fixed the issue. But it does suffer from minor inconvenience- you now have to care about the permissions on the dir you mount (it has to be accessible by the user in the container so you have to make sure you set valid permissions on it before).

Reply:
I see. I don't find the solution to have some hardcoded user and group id very satisfying.
I'm surprised there is no out-of-the-box solution for this provided by docker, the problem seem to be super common to me.
Making user and group id configurable would make this more operable I guess.

Anyway, do you intend to document the user which needs to have permissions on the project's directory?

Reply:
The only thing hardcoded is the default UID and GID, you can override them when running the container. Proper volume mounting has to be documented because with a bad configuration you can end up in permission errors.

Reply:
Thanks for you reply.

I actually understood that and found out how it worked after I made this comment.
Also, I noticed the uid of the current user was set but not the guid, which, on my box, led to create files with my user but group root (which should be impossible according the my box's set up). I dig and created #16